src/digodoc_lib/cmt.ml

```diff
@@ -64,12 +64,191 @@ let pp_val_desc ident pp vdesc =
         done;
         Format.fprintf pp "%s" (Buffer.contents buff)
 
-let getVals {cmi_sign; _} = 
-    let vals = List.filter (function | Sig_value _ -> true | _ -> false) cmi_sign in
-    List.map
-        (fun sign ->
-            match sign with
-            | Sig_value (ident, vald, _) -> 
-                infix (Ident.name ident), Format.asprintf "%a" (pp_val_desc ident) vald
-            | _ -> failwith "should not occur")
-        vals
\ No newline at end of file
+let getVals {cmi_sign; _} =
+    (*
+       For documentation on the types matched here see:
+        https://docs.ocaml.pro/docs/LIBRARY.ocamlcommon@ocaml-base-compiler.4.10.0/Types/index.html#type-signature_item
+      *)
+    List.filter_map (function
+            | Sig_value (ident, vald, _) ->
+                Some (infix (Ident.name ident), Format.asprintf "%a" (pp_val_desc ident) vald)
+            | _ -> None)
+        cmi_sign
+
+(** [standardise_output s] Removes and replaces some char occurences *)
+let standardise_output s =
+    String.map (function | '{' | '}' -> '\x00' | '\r' -> ' ' | '\t' -> ' ' | '\x0C' -> ' ' | n -> n) s
+
+(** [out_type fmt type] recursively prints a type declaration tree *)
+let rec out_type fmt sign =
+    let rec aux_out_type fmt =
+        let open Outcometree in
+        function
+            | Otyp_open -> ()
+            | Otyp_abstract -> ()
+            | Otyp_record list_of_fields ->
+                    Format.fprintf fmt "%a" (Format.pp_print_list
```

Comment:
You can remove the `Format.fprintf fmt "%a"` and call `pp_print_list` directly (same comment in a lot of places).